### PR TITLE
Remove udt from pk

### PR DESCRIPTION
**Pull request by aduprat**
This is a POC.

### CODE REVIEW

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java [hunk 1]
```diff
@@ -110,6 +114,27 @@ public Mailbox findMailboxByPath(MailboxPath path) throws MailboxException {
...
+            return mailboxPathDAO.retrieveId(path)
+                .thenCompose(cassandraIdOptional ->
+                    cassandraIdOptional
+                        .map(CassandraIdAndPath::getCassandraId)
+                        .map(this::retrieveMailbox)
+                        .orElse(CompletableFuture.completedFuture(Optional.empty())));
+        } catch (CompletionException e) {
+            if (e.getCause() instanceof InvalidQueryException) {
+                String errorMessage = e.getCause().getMessage();
+                if (StringUtils.containsIgnoreCase(errorMessage, VALUES_MAY_NOT_BE_LARGER_THAN_64_K)
```
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: This `if condition` needs to be extracted.

I would value having a `DriverExceptionHelper` class in the backend managing it ^^

**Comment by chibenwa** [comment]
```
[390b0d1a96bd17052608ebb9f647a2b0158f3083] Tests in error: 
[390b0d1a96bd17052608ebb9f647a2b0158f3083]   MailboxMapperTest.findMailboxWithPathLikeWithNullUserWithChildRegexShouldRetrieveChildren » Completion
[390b0d1a96bd17052608ebb9f647a2b0158f3083]   MailboxMapperTest.findMailboxWithPathLikeWithNullUserWithRegexShouldRetrieveCorrespondingMailbox » Completion
[390b0d1a96bd17052608ebb9f647a2b0158f3083]   MailboxMapperTest.hasChildrenWithNullUserShouldReturnFalseWhenNoChildrenExists » Completion
[390b0d1a96bd17052608ebb9f647a2b0158f3083]   MailboxMapperTest.hasChildrenWithNullUserShouldReturnTrueWhenChildrenExists » Completion
```

**Comment by chibenwa** [comment]
test this please

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/utils/DriverExceptionHelper.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.util.concurrent.CompletionException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.exception.TooLongMailboxNameException;
+
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+
+public class DriverExceptionHelper {
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/utils/DriverExceptionHelper.java]: Throws mailbox specific exceptions... :-/

**Comment by rouazana** [comment]
test fixed and pr rebased

**Comment by chibenwa** [comment]
test this please (no space left on device)

**Pull request closed without merging**